e2e-test-server/src/main/java/com/google/cloud/opentelemetry/endtoend/Server.java

```diff
@@ -100,45 +109,27 @@ public void onFailure(Throwable t) {
         MoreExecutors.directExecutor());
   }
 
-  /** Execute a scenario based on the incoming message from the test runner. */
-  public void handleMessage(PubsubMessage message, AckReplyConsumer consumer) {
-    if (!message.containsAttributes(Constants.TEST_ID)) {
-      consumer.nack();
-      return;
-    }
-    String testId = message.getAttributesOrDefault(Constants.TEST_ID, "");
-    if (!message.containsAttributes(Constants.SCENARIO)) {
-      respond(
-          testId,
-          Response.invalidArgument(
-              String.format("Expected attribute \"%s\" is missing", Constants.SCENARIO)));
-      consumer.ack();
-      return;
-    }
-    String scenario = message.getAttributesOrDefault(Constants.SCENARIO, "");
-    Request request = Request.make(testId, message.getAttributesMap(), message.getData());
-
-    // Run the given request/response cycle through a handler and respond with results.
-    Response response = Response.EMPTY;
-    try {
-      response = scenarioHandlers.handleScenario(scenario, request);
-    } catch (Throwable e) {
-      e.printStackTrace(System.err);
-      response = Response.internalError(e);
-    } finally {
-      respond(testId, response);
-      consumer.ack();
-    }
-  }
-
   /** Runs our server. */
   public static void main(String[] args) throws Exception {
-    try (Server server = new Server()) {
-      server.start();
-      // Docs for Subscriber recommend doing this to block main thread while daemon thread consumes
-      // stuff.
-      for (; ; ) {
-        Thread.sleep(Long.MAX_VALUE);
+    Server server = new Server();
+    if (Constants.SUBSCRIPTION_MODE.equals(Constants.SUBSCRIPTION_MODE_PULL)) {
+      try (PubSubServer pullServer = new PubSubPullServer(server)) {
```

Comment:
Nit: you can simplify this code by creating a new "factory function" for PubSub server type:
```java
  try (PubSubServer server = createPubSubServer(server)) {
    ...
  }


private PubSubServer createPubSubServer(Server server) throws Exception {
  if (Constants.SUBSCRIPTION_MODE.equals(Constants.SUBSCRIPTION_MODE_PULL)) {
     return new PubSubPullServer(server):
   }
   return new PubSubPushServer(Integer.parseInt(Constants.PUSH_PORT), server);
}
```